### PR TITLE
Update Notes Api with changes made to Base Api template

### PR DESCRIPTION
**Pull request by humulla**
## Link to JIRA ticket

[Jira Ticket](https://hackney.atlassian.net/jira/software/projects/MTTL/boards/47?selectedIssue=MTTL-432)

### *What is the problem we're trying to solve*

Updating the notes API so that it matched the BaseAPI which have been agreed by the team 

### *What changes have we introduced*

- [x] Correlation Id Middleware support
- [x] AWS X-Ray support
- [x] AWS DynamoDb support

#### _Checklist_
- [x] Added tests to cover all new production code
- [x] Added comments to the README 
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: NotesApi/Startup.cs [hunk 1]
```diff
@@ -115,12 +122,33 @@ private static void ConfigureDbContext(IServiceCollection services)
             var connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING");
 
             services.AddDbContext<DatabaseContext>(
-                opt => opt.UseNpgsql(connectionString));
+                opt => opt.UseNpgsql(connectionString).AddXRayInterceptor(true));
```
Comment (mikemoody-amido) [inline NotesApi/Startup.cs:125]: This entire function (`ConfigureDbContext`) can be removed.

File: NotesApi.Tests/V1/E2ETests/ExampleTest.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 namespace NotesApi.Tests.V1.E2ETests
 {
     //For guidance on writing integration tests see the wiki page https://github.com/LBHackney-IT/lbh-base-api/wiki/Integration-Tests
-    public class ExampleTest : IntegrationTests<Startup>
+    public class ExampleTest : DynamoDbIntegrationTests<Startup>
     {
 
     }
```
Comment (mikemoody-amido) [inline NotesApi.Tests/V1/E2ETests/ExampleTest.cs:7]: Do we need this class anymore?

File: NotesApi/Startup.cs [hunk 3]
```diff
@@ -115,12 +122,33 @@ private static void ConfigureDbContext(IServiceCollection services)
...
+                if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == Environments.Development)
+                {
+                    config.AddConsole();
+                }
+            });
         }
 
         private static void RegisterGateways(IServiceCollection services)
         {
             services.AddScoped<INotesApiGateway, NotesApiGateway>();
```
Comment (mikemoody-amido) [inline NotesApi/Startup.cs:150]: This line should be deleted.
And if it still exists, the `NotesApiGateway` class should also be deleted, as it is Postgres-specific.

**Review by mikemoody-amido — COMMENTED** [review]
You also need to move the terraform folders and add the necessary terraform files (see the person Api for an example).

File: docker-compose.yml [hunk 4]
```diff
@@ -9,7 +9,11 @@ services:
...
       - 3000:3000
     environment:
-      - CONNECTION_STRING=Host=dev-database;Port=5432;Database=testdb;Username=postgres;Password=mypassword
+      - DynamoDb_LocalMode=true
+      - DynamoDb_LocalServiceUrl=http://dynamodb-database:8000
+      - AWS_REGION=eu-west-2
+      - AWS_ACCESS_KEY_ID=local
+      - AWS_SECRET_ACCESS_KEY=local
     links:
       - dev-database
```
Comment (mikemoody-amido) [inline docker-compose.yml:18]: This should be `dynamodb-database` and you can remove the `dev-database` section directly below.

**Review by evak6979 — APPROVED** [review]

**Review by mikemoody-amido — APPROVED** [review]

**Pull request merged**